src/passes/annotateImplicits.ts

```diff
@@ -129,3 +136,47 @@ class ImplicitCollector extends ASTVisitor<Set<Implicits>> {
     return result;
   }
 }
+
+function extractImplicitFromStubs(node: FunctionDefinition, result: Set<Implicits>) {
+  const cairoCode = getDocString(node.documentation);
+  assert(cairoCode !== undefined);
+  const funcSignature = cairoCode.match(/func .+\{(.+)\}/);
+  if (funcSignature === null) return;
+
+  // implicits -> impl1 : type1, impl2, ..., impln : typen
+  const implicits = funcSignature[1];
+
+  // implicitsList -> [impl1 : type1, impl2, ...., impln : typen]
+  const implicitsList = [...implicits.matchAll(/[A-Za-z][A-Za-z_: 0-9]*/g)].map((w) => w[0]);
+
+  // implicitsNameList -> [impl1, impl2, ..., impln]
+  const implicitsNameList = implicitsList.map((i) => i.match(/[A-Za-z][A-Za-z_0-9]*/));
+  if (!notContainsNull(implicitsNameList)) return;
+
+  // Check that implicits are valid and add them to result
+  implicitsNameList.forEach((i) => {
+    const impl = i[0];
+    if (!elementIsImplicit(impl)) {
+      throw new TranspileFailedError(
+        `Implicit ${impl} defined on function stub (${printNode(node)}) is not known`,
+      );
+    }
+    result.add(impl);
+  });
+}
+
+function elementIsImplicit(e: string): e is Implicits {
+  const knownImplicits = [
+    'bitwise_ptr',
```

Comment:
The list of known implicits is already in `src/utils/implicits.ts` you can import from there and avoid code duplication